audit-service/src/contract-tests/java/uk/gov/gchq/palisade/contract/audit/kafka/KafkaContractTest.java

```diff
@@ -221,20 +249,84 @@ void testRestEndpointForGoodSuccessMessage() {
+        // Add a message to the 'error' topic
+        // The ContractTestData.REQUEST_TOKEN maps to partition 0 of [0, 1, 2], so the akka-test yaml connects the consumer to only partition 0
+        final Stream<ProducerRecord<String, JsonNode>> requests = ContractTestData.BAD_ERROR_MESSAGE_NODE_FACTORY.get().limit(1L);
+
+        // When - we write to the input
+        ProducerSettings<String, JsonNode> producerSettings = ProducerSettings
+                .create(akkaActorSystem, new StringSerializer(), new RequestSerializer())
+                .withBootstrapServers(KafkaInitializer.kafka.getBootstrapServers());
+
+        Source.fromJavaStream(() -> requests)
+                .runWith(Producer.plainSink(producerSettings), akkaMaterializer);
+
+        TimeUnit.SECONDS.sleep(2);
+
+        // Then check an "Error-..." file has been created
+        Set<File> expectedFileSet = Arrays.stream(new File(".").listFiles())
+                .filter(file -> file.getName().startsWith("Error"))
+                .collect(Collectors.toSet());
+        Set<File> actualFileSet = Arrays.stream(new File(auditServiceConfigProperties.getErrorDirectory()).listFiles())
+                .filter(file -> file.getName().startsWith("Error"))
+                .collect(Collectors.toSet());
+
+        assertAll(
+                () -> assertThat(actualFileSet).as("Check at least 1 'Error' file has been created").isNotEmpty(),
+                () -> assertThat(actualFileSet).as("Check the 'Error' files are the same").containsAll(expectedFileSet)
+        );
+    }
+
+    @Test
+    @DirtiesContext
+    void testFailedSuccessDeserialization() throws InterruptedException {
+        // Add a message to the 'success' topic
+        // The ContractTestData.REQUEST_TOKEN maps to partition 0 of [0, 1, 2], so the akka-test yaml connects the consumer to only partition 0
+        final Stream<ProducerRecord<String, JsonNode>> requests = ContractTestData.BAD_SUCCESS_MESSAGE_NODE_FACTORY.get().limit(1L);
+
+        // When - we write to the input
+        ProducerSettings<String, JsonNode> producerSettings = ProducerSettings
+                .create(akkaActorSystem, new StringSerializer(), new RequestSerializer())
+                .withBootstrapServers(KafkaInitializer.kafka.getBootstrapServers());
+
+        Source.fromJavaStream(() -> requests)
+                .runWith(Producer.plainSink(producerSettings), akkaMaterializer);
+
+        TimeUnit.SECONDS.sleep(2);
+
+        // Then check an "Success-..." file has been created
+        Set<File> expectedFileSet = Arrays.stream(new File(".").listFiles())
+                .filter(file -> file.getName().startsWith("Success"))
+                .collect(Collectors.toSet());
+        Set<File> actualFileSet = Arrays.stream(new File(auditServiceConfigProperties.getErrorDirectory()).listFiles())
+                .filter(file -> file.getName().startsWith("Success"))
+                .collect(Collectors.toSet());
```

Comment:
This test boils down to `assert auditServiceConfigProperties.getErrorDirectory() == "."`
Suggest find a better way to test it, but not sure how.

Perhaps drop the actual vs expected and change to an old vs new?
i.e. before the test there was this set of files, after there was the same set plus one more?

Reply:
Updated the assert to check that an extra file has been created rather than checking the files sets are the same.